commit 72549bef582001b6dbec2ecb83e094ccb57d72c8
Author: Mark Featherston <mark@embeddedarm.com>
Date:   Wed May 18 16:33:53 2016 -0700

    New chips are requiring longer delays before we start sending SPI.  Some new boards always return "FPGA_DONE never asserted" with the previous delays.   The datasheet specifies 300us, we were waiting 800us, and testing shows it start working at 900us.  We're moving to 2ms.

Upstream-Status: Backport

Signed-off-by: Michal Mazurek <michal@resin.io>

diff --git a/common/cmd_ice40.c b/common/cmd_ice40.c
index c292eb5..3f45d4d 100644
--- a/common/cmd_ice40.c
+++ b/common/cmd_ice40.c
@@ -24,7 +24,7 @@ static int do_ice40_load(cmd_tbl_t *cmdtp, int flag, int argc, char * const argv
 	gpio_direction_output(CONFIG_ICE40_CS, 0); // spi cs# low
 	udelay(1); // at least 200ns
 	gpio_set_value(CONFIG_ICE40_FPGA_RESET, 1); // reset high
-	udelay(800);
+	mdelay(2);
 
 	ret_val = spi_xfer(slave, len * 8, (void *)data, NULL, 0);
 
